base/indices.jl

```diff
@@ -271,8 +271,8 @@ Custom array types may specialize `to_index(::CustomArray, i)` to provide
 special indexing behaviors. Note that some index types (like `Colon`) require
 more context in order to transform them into an array of indices; those get
 converted in the more complicated `to_indices` function. By default, this
-simply calls the generic `to_index(i)`. This must return either an `Int` or an
-`AbstractArray` of scalar indices that are supported by `A`.
+simply calls the generic `to_index(i)`. This must return either a subtype of `Integer`
```

Comment:
`Bool <: Integer`, but we currently give an error for `A[true]`, which seems prudent…

Reply:
It still errors though, you explicitly need to override `Base.to_index` to actually preserve the index.